pkg/reconciler/pipelinerun/resources/pipelineref.go

```diff
@@ -118,16 +118,32 @@ func GetVerifiedPipelineFunc(ctx context.Context, k8s kubernetes.Interface, tekt
 			source = s.URI
 		}
 		logger := logging.FromContext(ctx)
-		if config.CheckEnforceResourceVerificationMode(ctx) || config.CheckWarnResourceVerificationMode(ctx) {
-			if err := trustedresources.VerifyPipeline(ctx, p, k8s, source, verificationpolicies); err != nil {
-				if config.CheckEnforceResourceVerificationMode(ctx) {
-					logger.Errorf("GetVerifiedPipelineFunc failed: %v", err)
-					return nil, nil, fmt.Errorf("GetVerifiedPipelineFunc failed: %w: %v", trustedresources.ErrResourceVerificationFailed, err)
-				}
+		// skip the verification if no policies are found and trusted-resources-verification-no-match-policy is set to ignore
+		if len(verificationpolicies) == 0 {
+			if config.CheckWarnVerificationNoMatchPolicy(ctx) {
+				// log warning and don't return error when trusted-resources-verification-no-match-policy is set to warn
 				logger.Warnf("GetVerifiedPipelineFunc failed: %v", err)
 				return p, s, nil
```

Comment:
Not sure if you intended on covering this in this PR or not. I don't think this does anything for setting the status on the object.

Reply:
We don't update status in this PR. I plan to add it in a followup pr